app/models/links.rb

```diff
@@ -35,8 +35,18 @@ def links
   delegate :keys, to: :links
 
   def unique_links
-    links.uniq do |_short, long|
-      send(long)
+    seen_urls = {}
+    links.select do |short, long|
+      url = send(long)
+      # always include 'code' (source_code_uri) even if URL is duplicate
+      if short == "code"
+        true
+      elsif seen_urls[url]
+        false
+      else
+        seen_urls[url] = true
+        true
+      end
```

Comment:
This is slightly confusing to me since there's a lot of booleans and conditionals, how about this?

Have `REQUIRED_INDEXED_LINK_KEYS`
```suggestion
    unique_links = links.uniq do |_short, long|
      send(long)
    end.to_h

    # always include certain URLs even if URL is a duplicate
    links.select do |short, _long|
      unique_links.key?(short) || REQUIRED_INDEXED_LINK_KEYS.include?(short)
    end
```